src/mscorlib/src/System/Reflection/AssemblyNameLexer.cs

```diff
@@ -0,0 +1,137 @@
+        // use GetNext(out String) instead.
+        //
+        internal Token GetNext()
+        {
+            String ignore;
+            return GetNext(out ignore);
+        }
+
+        //
+        // Return the next token in assembly name. If the result is DisplayNameToken.String, 
+        // sets "tokenString" to the tokenized string.
+        //
+        internal Token GetNext(out String tokenString)
+        {
+            tokenString = null;
+            while (Char.IsWhiteSpace(_chars[_index]))
+                _index++;
+
+            char c = _chars[_index++];
+            if (c == 0)
+                return Token.End;
+            if (c == ',')
+                return Token.Comma;
+            if (c == '=')
+                return Token.Equals;
+
+            StringBuilder sb = new StringBuilder();
+
+            char quoteChar = (char)0;
+            if (c == '\'' || c == '\"')
+            {
+                quoteChar = c;
+                c = _chars[_index++];
+            }
+
+            for (;;)
+            {
+                if (c == 0)
+                {
+                    _index--;
+                    break;  // Terminate: End of string (desktop compat: if string was quoted, permitted to terminate without end-quote.)
+                }
+
+                if (quoteChar != 0 && c == quoteChar)
+                    break;  // Terminate: Found closing quote of quoted string.
+
+                if (quoteChar == 0 && (c == ',' || c == '='))
+                {
+                    _index--;
+                    break;  // Terminate: Found start of a new ',' or '=' token.
+                }
+
+                if (quoteChar == 0 && (c == '\'' || c == '\"'))
+                    throw new FileLoadException();  // Desktop compat: Unescaped quote illegal unless entire string is quoted.
+
+                if (c == '\\')
+                {
+                    c = _chars[_index++];
+                    bool matched = false;
+                    foreach (KeyValuePair<char, String> kv in EscapeSequences)
+                    {
+                        if (c == kv.Key)
+                        {
+                            matched = true;
+                            sb.Append(kv.Value);
+                            break;
+                        }
+                    }
+                    if (!matched)
+                        throw new FileLoadException();  // Unrecognized escape
+                }
+                else
+                {
+                    sb.Append(c);
+                }
+
+                c = _chars[_index++];
+            }
+
+            tokenString = sb.ToString();
+            if (quoteChar == 0)
+                tokenString = tokenString.Trim(); // Unless quoted, whitespace at beginning or end doesn't count.
+            return Token.String;
+        }
+
+        internal static KeyValuePair<char, String>[] EscapeSequences =
```

Comment:
Including all of AssemblyNameLexer just to get EscapeSequences seems excesssive (unless you decide to throw the whole set into the shared partition.)